### PR TITLE
PP-4846 Go live task list: show next steps box at right point

### PR DESCRIPTION
**Pull request by alexbishop1**
Show the next steps box saying we are creating the live service when the current go live stage is `AGREED_TERMS_STRIPE`, `AGREED_TERMS_WORLDPAY`, `AGREED_TERMS_SMARTPAY` or `AGREED_TERMS_EPDQ` and not when it’s `LIVE`.

### CODE REVIEW

File: app/controllers/request-to-go-live/index/get.controller.js [hunk 1]
```diff
@@ -1,16 +1,82 @@
...
+]
+
+const agreedToTerms = [
+  TERMS_AGREED_STRIPE,
+  TERMS_AGREED_WORLDPAY,
+  TERMS_AGREED_SMARTPAY,
+  TERMS_AGREED_EPDQ
+].concat(live)
+
+const chosenHowToProcessPayments = [
```
Comment (sfount) [inline app/controllers/request-to-go-live/index/get.controller.js]: (This certainly shouldn't be blocking - it depends on how often this pattern will continue in the future. tl;dr define constants and groups once)

It seems like we're pulling and categorising these constant out of this object map in a number of different controllers. Would it make sense to move this logic over to the helper object/ model? 

Potential pros: 
* controllers don't have to duplicate code 
* controllers are simple and easy to read (although this code is very clear!) 

Something like: 
```js
const goLiveStage = require('goLiveStage')

const goLiveIndexHandler = function goLiveIndexHandler (req, res, next) { 
  const { currentGoLiveStage } = req.service
  const goLiveProcessStatus = goLiveStage.status(currentGoLiveStage)

  // ensure cookies set in previous session aren't propagated
  delete req.session.pageData.requestToGoLive.index

  return response(req, res, 'request-to-go-live/index', pageData)
}

module.exports = goLiveIndexHandler
```

Or  a middle ground would be to do the categorising only once in the helper/ model and then use those for logic in the controllers: 

```js
const { agreedToTermsStates, sessionIsAgreeingToTerms } = require('goLiveStage') 

// check if my status is in the agreed terms
agreedToTermsStates.includes(currentGoLiveStage) 

// or 
sessionIsAgreeingToTerms(currentGoLiveStage)
```

Files that also do this: 
https://github.com/alphagov/pay-selfservice/blob/master/app/controllers/request-to-go-live/choose-how-to-process-payments/post.controller.js#L16
https://github.com/alphagov/pay-selfservice/blob/master/app/controllers/request-to-go-live/agreement/get.controller.js#L12

File: app/views/request-to-go-live/index.njk [hunk 2]
```diff
@@ -7,15 +7,15 @@
 {% endblock %}
 
 {% block mainContent %}
-  {% if currentService.currentGoLiveStage === goLiveStage.LIVE %}
```
Comment (DanailMinchev) [inline app/views/request-to-go-live/index.njk]: Moving the logic to the controller looks nice 👍

File: app/controllers/request-to-go-live/index/get.controller.js [hunk 3]
```diff
@@ -1,16 +1,82 @@
...
+const live = [
+  LIVE
+]
+
+const agreedToTerms = [
+  TERMS_AGREED_STRIPE,
+  TERMS_AGREED_WORLDPAY,
+  TERMS_AGREED_SMARTPAY,
+  TERMS_AGREED_EPDQ
+].concat(live)
```
Comment (DanailMinchev) [inline app/controllers/request-to-go-live/index/get.controller.js]: You can use `Destructuring assignment`, for example:
```
const agreedToTerms = [
  TERMS_AGREED_STRIPE,
  TERMS_AGREED_WORLDPAY,
  TERMS_AGREED_SMARTPAY,
  TERMS_AGREED_EPDQ,
  ...live
]
```

We use it in https://github.com/alphagov/pay-selfservice/blob/884eaf4eaf9b1b1796b65911f288110cef1f0959/app/routes.js#L174 and https://github.com/alphagov/pay-selfservice/blob/884eaf4eaf9b1b1796b65911f288110cef1f0959/app/utils/json_to_csv.js#L43 and I think it's the latest syntax which we discussed

File: app/controllers/request-to-go-live/index/get.controller.js [hunk 4]
```diff
@@ -1,16 +1,82 @@
...
+
+const chosenHowToProcessPayments = [
+  CHOSEN_PSP_STRIPE,
+  CHOSEN_PSP_WORLDPAY,
+  CHOSEN_PSP_SMARTPAY,
+  CHOSEN_PSP_EPDQ
+].concat(agreedToTerms)
```
Comment (DanailMinchev) [inline app/controllers/request-to-go-live/index/get.controller.js]: same for other `concat`s

File: app/views/policy/document-downloads/contract_for_non_crown_bodies.njk [hunk 5]
```diff
@@ -5,7 +5,7 @@
 {% endblock %}
 
 {% block policyDescription %}
-  If you're a non-crown body, the contract terms apply when you use GOV.UK Pay.
+  If you’re a non-crown body, the contract terms apply when you use GOV.UK Pay.
```
Comment (DanailMinchev) [inline app/views/policy/document-downloads/contract_for_non_crown_bodies.njk]: Sorry, a bit picky: this can be in a different commit, thanks for fixing that according to latest content rules 👍

File: app/controllers/request-to-go-live/index/get.controller.js (hunk 3 above)
Comment (alexbishop1) [reply app/controllers/request-to-go-live/index/get.controller.js]: Ooh, I like that!

File: app/views/policy/document-downloads/contract_for_non_crown_bodies.njk (hunk 5 above)
Comment (alexbishop1) [reply app/views/policy/document-downloads/contract_for_non_crown_bodies.njk]: Yeah, it’s unrelated but I thought I’d sneak it in because it’s tiny

File: test/cypress/integration/request-to-go-live/index_spec.js [hunk 6]
```diff
@@ -60,14 +60,15 @@ describe('Request to go live: index', () => {
       cy.visit(requestToGoLivePageUrl)
 
       cy.get('h1').should('contain', 'Request to go live')
+      cy.get('h1 + p').should('contain', 'Complete these steps to start taking real payments')
 
-      cy.get('#request-to-go-live-step-1 > h3').should('exist')
```
Comment (DanailMinchev) [inline test/cypress/integration/request-to-go-live/index_spec.js]: In the original code it is not obvious here why we assert on existence and not on content, the idea was not to depend on text/content of each step in case there are minor content changes / or major which should not break the test. That is why I prefer to rely on ids not on content.
But, after discussion with @alexbishop1 I understand that it is not clear why we have this assertions and which step is what.
We discussed to change the ids to be more descriptive or to leave the content as it is highly coupled with implementation (as steps).

Thanks

Comment (alexbishop1) [reply test/cypress/integration/request-to-go-live/index_spec.js]: Just pushed a commit that asserts less on text and makes the IDs more descriptive

File: app/controllers/request-to-go-live/index/get.controller.js (hunk 1 above)
Comment (alexbishop1) [reply app/controllers/request-to-go-live/index/get.controller.js]: Yes, something to consider for the future
